ROS 2 / Humble - porting

Iranaphor opened the pull request:
Includes:
- Changes to package hierarchy to conform to ROS2 standards.
- Updates to dependencies to reference ROS2 equivalents.

Much is left to port, however I thought it best to open a (WIP) PR now.

Currently I have not yet addressed topological_utils meaning topological_rviz_tools is also uncompilable. By skipping topological_rviz_tools topological_utils, the remaining 3 packages will build successfully. However containing code will still require further porting. 

`colcon build --symlink-install --packages-skip topological_rviz_tools topological_utils`

A noteworthy change which was required, is to some msg files which utilised the field name "namespace". The rosidl_generator_cpp is not able to compile msg files which utilise keywords from c. I have renamed these instances instead, to "namespace_name".

Iranaphor commented:
The build process should work now, however note that none of the ROS1 CPP files have been ported yet, (and they are commented out in their respective `CMakeLists.txt` file to avoid a build crash). The ROS1 Python files have also not been ported yet, nor have any ROS1 Launch files.

The Service files from `topological_utils` have been moved into `topological_navigation_msgs` due to some conflicts between `rosidl_generate_interfaces` and `ament_python_install_package` which I was not able to fully solve.It seemed to make sense given we have a msgs package already available and included, so now `topological_utils`, is now looking to be a full python package rather than a combined one. Depending on feedback from @marc-hanheide (+others?), I might recommend also moving messages from `bayesian_topological_navigation` to `topological_navigation_msgs`.

Due to my unfamiliarity with ROSCPP and RCLCPP, I am going to avoid updating the `topological_rviz_tools` package myself, and leave this for someone more experienced.

marc-hanheide commented:
Awesome stuff, @Iranaphor !

I fully agree that the messages should all be moved.

Great you took this on and made it work. I’m also not familiar with CPP in ROS2 yet, so we need to see how we migrate the `topological_rviz_tools`.

marc-hanheide commented on topological_rviz_tools/CMakeLists.txt:
```diff
@@ -1,108 +1,212 @@
+#
+## Construct the library and define its dependencies
+## Declare files to be included under the plugin
+#set(rviz_files
```
So, @Iranaphor this is where the C++ code needs to be re-added once it is changed to ROS2, right?

Iranaphor replied:
Yes, by uncommenting the entire code block, the compiler crashes trying to build roscpp elements. More might be needed beyond to fully install the components but we can find that out as we progress further.

marc-hanheide commented on topological_navigation/package.xml:
```diff
@@ -1,10 +1,10 @@
 
-
   <maintainer email="abinch@sagarobotics.com">Adam Binch</maintainer>
 
```
I think we need to add you or somebody else from UoL as a maintained here @Iranaphor (you can also list me)

marc-hanheide commented on topological_navigation_msgs/package.xml:
```diff
@@ -1,27 +1,29 @@
   <license>MIT</license>
 
-  <author>Adam Binch</author>
-  <author>Jaime Pulido Fentanes</author>
```
Have you removed the authors on purpose? I’d leave them in, but add yourself, @Iranaphor

marc-hanheide replied:
Same for the maintainers

Iranaphor replied:
There was a lot of copying and pasting files so i must have missed some. Now it is building, I'll go through and add back in the original authors and maintainers (+ myself).

marc-hanheide commented on topological_navigation_msgs/srv/UpdateEdgeConfig.srv:
```diff
@@ -1,5 +1,5 @@
 string edge_id
-string namespace
+string namespace_name
```
This is an annoying thing… but I see why it’s needed. Great you spotted that problem. It may require us to go through all our code searching for problems stemming from this change.

marc-hanheide reviewed (approved):
I approve this, given that it is a massive step forward and means we can start the migration to ROS2. I think we can merge this and start building an actual ROS2 system. I’m sure we’ll see more issues once we start using it, but this is great progress.

Iranaphor commented:
Combining the message files, there were two srv files by the same name `AddEdge`, the instance in the `topological_rviz_tools` package, I have renamed (along with its uses within the stack) to `AddEdgeRviz` during the centralising to topological_navigation_msgs.

I have fixed the authorships to reflect the ROS1 HEAD, and have now synced the packages to all be version `3.0.0` to coincide with the major update to ROS2. Including myself as a maintainer on all packages and Marc on `topological_navigation`. The `setup.py` files have also had their maintainers updated to reflect the first maintainer in their respective `package.xml` files.

I have noticed that, unlike the other packages, bayesian_topological_localisation's lisence is BSD, rather than MIT. Is this intentional of should it coincide with the rest?
